frontend/src/Utils/api-discovery/getResources.test.ts

```diff
@@ -0,0 +1,306 @@
+  });
+  test('groupVersion without `/`', () => {
+    const result = defineModels({
+      ...apiResouceList,
+      groupVersion: 'foo',
+    });
+    expect(result).toStrictEqual([
+      {
+        apiVersion: 'foo',
+        kind: 'bar',
+        namespaced: undefined,
+        verbs: ['get'],
+        shortNames: undefined,
+        plural: 'foo',
+        crd: true,
+        abbr: 'ABR',
+        labelPlural: 'ABRs',
+        path: 'foo',
+        id: undefined,
+        label: 'bar',
+      },
+    ]);
+  });
+  test('filter out models with `/` in name', () => {
+    const result = defineModels({
+      ...apiResouceList,
+      resources: [
+        {
+          name: 'foo/bar',
+          kind: 'bar',
+          verbs: ['get'],
+        },
+      ],
+    });
+    expect(result).toStrictEqual([]);
+  });
+});
+
+describe('batchResourcesRequest', () => {
+  const data: APIResourceList = {
+    kind: 'APIResourceList',
+    apiVersion: 'v1',
+    groupVersion: 'foo/bar',
+    plural: 'Foo',
+    resources: [
+      {
+        name: 'foo',
+        kind: 'bar',
+        verbs: ['get'],
+      },
+    ],
+  };
+  const model = {
+    apiGroup: 'foo',
+    apiVersion: 'bar',
+    kind: 'bar',
+    namespaced: undefined,
+    verbs: ['get'],
+    shortNames: undefined,
+    plural: 'foo',
+    crd: true,
+    abbr: 'ABR',
+    labelPlural: 'ABRs',
+    path: 'foo',
+    id: undefined,
+    label: 'bar',
+  };
+  test('should calculate resources', async () => {
+    common.commonFetchJSON.mockReturnValueOnce(data);
+    const result = await Promise.all(batchResourcesRequest(['/foo']));
+    expect(result).toStrictEqual([
+      {
+        allResources: ['foo'],
+        safeResources: ['foo'],
+        adminResources: [],
+        configResources: [],
+        clusterOperatorConfigResources: [],
+        namespacedSet: new Set(),
+        models: [model],
+      },
+    ]);
+  });
+  test('should gracefully handle empty resources', async () => {
+    const currData = { ...data };
+    delete currData.resources;
+    common.commonFetchJSON.mockReturnValueOnce(currData);
+    const result = await Promise.all(batchResourcesRequest(['/foo']));
+    expect(result).toStrictEqual([
+      {
+        allResources: [],
+        safeResources: [],
+        adminResources: [],
+        configResources: [],
+        clusterOperatorConfigResources: [],
+        namespacedSet: new Set(),
+        models: [],
+      },
+    ]);
+  });
+  test('should calculate admin resources', async () => {
+    const currData = { ...data };
+    currData.resources = [
+      {
+        name: 'roles/foo',
+        kind: '',
+        verbs: ['get'],
+      },
+      {
+        name: 'foo',
+        kind: 'bar',
+        verbs: ['get'],
+      },
+    ];
+    common.commonFetchJSON.mockReturnValueOnce(currData);
+    const result = await Promise.all(batchResourcesRequest(['/foo']));
+    expect(result).toStrictEqual([
+      {
+        allResources: ['foo', 'roles/foo'],
```

Comment:
NIT and FWIW - since the structure of this asserted array is similar for each test (with one or two properties changed) - we could put it as a const in the before each setup and just change the fields we need to per test to avoid the duplication.